### PR TITLE
docker all the things

### PR DESCRIPTION
**Pull request by miltontony**

### CODE REVIEW

**Comment by jerith** [comment]
Can we split the celery stuff out into a separate branch?

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+set -e
+
+SUPERVISOR_APP_NAME=$1
+APP_WSGI=$2
+APP_PORT_NUMBER=$3
+DJANGO_SETTINGS_MODULE="${APP_WSGI:0:${#APP_WSGI}-5}.settings"
```
Comment (jerith) [inline docker-entrypoint.sh]: Why do these things need to be configurable? They're all set in the dockerfile's `ENTRYPOINT`, which means they can't be overridden.

**Comment by jerith** [comment]
If we remove the (almost certainly unnecessary) configurability in the configs, we can stop dynamically generating those and make them files that we copy into the image instead.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM python:2.7.10
+
+RUN apt-get update && apt-get install -y \
+	redis-server nginx
+
+RUN service nginx start`
```
Comment (smn) [inline Dockerfile]: What's this backtick doing here?

**Comment by smn** [comment]
I know you took a bunch of this from my garelay thing but my gut feeling is that sooner rather than later we're going to want to make this Docker container only about running the mc2 code, not supervisord and all the other things. I took this approach for garelay because I needed a single thing that had no outside requirements for deploying over satellite links. 

I'd suggest you work towards landing this first and then follow up with the working to split it out for when you next make time.

**Comment by jerith** [comment]
> [...] my gut feeling is that sooner rather than later we're going to want to make this Docker container only about running the mc2 code, not supervisord and all the other things.

Agreed. One of the reasons we're doing it like this for now is that we don't have the machinery we need for multiple interacting containers in seed-stack yet, and we'd like to get MC2 running there without blocking on all the other bits.

File: Dockerfile (hunk 2 above)
Comment (miltontony) [inline Dockerfile]: :) dirt on your screen I think..

**Comment by jerith** [comment]
I'd be happier with config files that we copy into the image rather than a giant shell script that creates them at startup. Also, is there a particular reason not to have all the supervisor configs in one file?

**Comment by miltontony** [comment]
@jerith ready for re-review

**Comment by jerith** [comment]
:+1: although I can't seem to connect to nginx in the container when I build and run it locally.

**Comment by miltontony** [comment]
\o/

**Pull request merged**